src/lib.rs

```diff
@@ -41,1346 +53,988 @@ use std::{
-        };
-        equation_test_template::<F, PC>(info)
+        let comms = commitments.into_iter().collect::<Vec<_>>();
+        let query_sets = query_sets.into_iter().collect::<Vec<_>>();
+        let values = values.into_iter().collect::<Vec<_>>();
+        let multi_point_proofs = multi_point_proofs.into_iter().collect::<Vec<_>>();
+        let states = states.into_iter().collect::<Vec<_>>();
+
+        // Perform succinct verification of all the proofs and collect
+        // the xi_s and the GFinal_s into DLogAccumulators
+        let succinct_time = start_timer!(|| "Succinct verification of proofs");
+
+        // TODO: into_par_iter should be used
+        let (accumulators, failed_checks): (
+            Vec<Result<_, Self::Error>>,
+            Vec<Result<_, Self::Error>>,
+        ) = comms
+            .into_iter()
+            .zip(query_sets)
+            .zip(values)
+            .zip(multi_point_proofs)
+            .zip(states)
+            .map(
+                |((((commitments, query_set), values), multi_point_proof), state)| {
+                    let mut fs_rng = Self::RandomOracle::default();
+                    fs_rng.set_state(state.clone());
+
+                    // Perform succinct check of i-th proof
+                    let verfifier_state = Self::succinct_multi_point_multi_poly_verify(
+                        vk,
+                        commitments,
+                        query_set,
+                        values,
+                        multi_point_proof,
+                        &mut fs_rng,
+                    )?;
+
+                    if verfifier_state.is_none() {
+                        Err(Error::FailedSuccinctCheck)?
+                    }
+
+                    Ok(verfifier_state.unwrap())
+                },
+            )
+            .partition(Result::is_ok);
+        end_timer!(succinct_time);
+
+        if !failed_checks.is_empty() {
+            Err(Error::FailedSuccinctCheck)?
+        }
+
+        let accumulators = accumulators
+            .into_iter()
+            .map(Result::unwrap)
+            .collect::<Vec<_>>();
+
+        Ok(accumulators)
     }
+}
 
-    pub fn two_equation_test<F, PC>(negative_type: Option<NegativeType>) -> Result<(), PC::Error>
-    where
-        F: Field,
-        PC: PolynomialCommitment<F>,
-    {
-        let info = TestInfo {
-            num_iters: 100,
-            max_degree: None,
-            supported_degree: None,
-            num_polynomials: 2,
-            enforce_degree_bounds: false,
-            max_num_queries: 1,
-            num_equations: Some(2),
-            segmented: false,
-            negative_type,
-            ..Default::default()
-        };
-        equation_test_template::<F, PC>(info)
+/// Evaluate the given polynomials at `query_set`.
+pub fn evaluate_query_set<'a, F: Field>(
+    polys: impl IntoIterator<Item = &'a LabeledPolynomial<F>>,
+    query_set: &QuerySet<'a, F>,
+) -> Evaluations<'a, F> {
```

Comment:
What is the use of this function, if we have the more expressive `fn evaluate_query_set_to_vec()`?

Reply:
Actually it is not used anywhere, I've removed it